docs/0___preamble.adoc

```diff
@@ -93,68 +93,65 @@ This document defines the Functional Mock-up Interface (FMI),
+
+For both Co-Simulation and Hybrid Co-Simulation the master algorithm is shielded from how the subsystem FMU advances time internally.
+For example, FMUs containing ODEs and exposing either of the Co-Simulation interfaces require to include an ODE solver inside the FMU to internally advance time between the communication points.
+Or, for controller code packaged as FMU, an internal scheduling algorithm will trigger tasks at the correct time and order while advancing time to the next communication point or event.
+
+_FMI for Model Exchange (<<fmi-for-model-exchange>>)_
+
+The Model Exchange interface exposes an ODE to an external solver of an importing tool's master algorithm.
+Models are described by differential, algebraic and discrete equations with time-, state- and step-events.
+That master algorithm, usually an ODE or DAE solver, like DASSL, is responsible to advance time, set states and handle events etc.
+
+_FMI for Scheduled Execution Simulation_
+
+The Scheduled Execution Simulation interface exposes individual parts or tasks of a control algorithm to be called from an external scheduler of an importing tool's master algorithm.
+That master algorithm scheduler is responsible to advance time and trigger execution of synchronous and asynchronous controller tasks.
+
+In many ways, the Scheduled Execution Simulation interface is the equivalent of the Model Exchange interface: the first externalizes a scheduling algorithm usually found in a controller algorithm and the second interface externalizes the ODE/DAE solver.
+
+image::images/fmi-types-overview.svg[width=50%, align="center"]
+
 _FMI Common Concepts (<<fmi-common-concepts>>)_
 
 The two interface standards have many parts in common.
-In particular, it is possible to utilize several
-instances of a model and/or a co-simulation tool and to connect them together.
-The interfaces are independent of the target environment because no header files
-are used that depend on the target environment
-(with exception of the data types of the target platform).
+In particular, it is possible to utilize several instances of a model and/or a co-simulation tool and to connect them together.
+The interfaces are independent of the target environment because no header files are used that depend on the target environment (with exception of the data types of the target platform).
 This allows generating one dynamic link library that can be utilized in any environment on the same platform.
-A model, a cosimulation slave or the coupling part of a tool,
-is distributed in one ZIP file called FMU (Functional Mockup Unit) that contains several files:
+A model, a cosimulation slave or the coupling part of a tool, is distributed in one ZIP file called FMU (Functional Mockup Unit) that contains several files:
```

Comment:
above instead of "salve" the expression "simulation subsystem" is used. So maybe co-simulation subsystem or sub model maybe used here instead of  cosimulation slave!?